test/ClangImporter/availability_non_runtime_protocol_objc_protocol.swift

```diff
@@ -0,0 +1,50 @@
+	
+// REQUIRES: objc_interop
+// REQUIRES: foundation
+
+// RUN: %target-swift-frontend -enable-objc-interop -typecheck -verify  -import-objc-header %S/Inputs/availability_non_runtime_protocol_objc_protocol.h %s
+
+import Foundation
+
+@objc class Wrapper : NSObject {
+  @objc public func runMutations2(_ mutations: [ObjCRuntimeProtocolUsedinSwift]) {
+    for _ in mutations {
+    }
+  }
+
+  public func foo(){
+    let _ : ObjCNonRuntimeProtocolUsedinSwift
+  }
+
+  @objc public func runMutations(_ mutations: [ObjCNonRuntimeProtocolUsedinSwift]) { // expected-error {{Objective-C protocol 'ObjCNonRuntimeProtocolUsedinSwift' has the non_runtime_protocol attribute and its metadata is not available in Swift}}
+    for _ in mutations {
+    }
+  }
+
+  public func iterate(collection mutations: some Collection<any ObjCNonRuntimeProtocolUsedinSwift>) { 
+      for m in mutations {
+          print(typeOf(anyNRP: m)) 
```

Comment:
@rjmccall  I moved the check to `checkTypeMetadataAvailabilityInternal`, but I am still a bit confused with what scenarios we should throw the diagnostics, e.g. is it expected we don't need metadata for this call at line 26, while we need metadata for the next call in line 27? Or are we missing this case in `checkTypeMetadataAvailabilityInternal`